src/verify_cert.rs

```diff
@@ -553,6 +553,7 @@ mod tests {
         ));
     }
 
+    #[ignore] // Runs slow from the inflated signature budget - opt-in with `cargo test -- --ignored`
```

Comment:
Alternatively: we could delete this test outright. Personally, I think `test_too_many_path_calls_mini` does a decent job of testing the mechanics of the budget without needing to use the real default budget value at the cost of additional runtime. WDYT? Now that it's ignored it also doesn't hurt much to leave it :shrug:

Reply:
I don't have a strong opinion, but killing it in favor of its mini sibling seems okay to me.

Reply:
@ctz Do you have a pref here?

Reply:
If we can get the same benefit from the faster test, let's just keep that. Ideally the tests of the default budget would be integration-level ones to be 100% sure that budget is being applied to users of the public API; though I realise it's quite annoying to share test code between integration and unit tests.

Reply:
> If we can get the same benefit from the faster test, let's just keep that.

:+1: 

> Ideally the tests of the default budget would be integration-level ones to be 100% sure that budget is being applied to users of the public API; though I realise it's quite annoying to share test code between integration and unit tests.

I think the issue is less about sharing test code and more that we're using the crate internal `build_chain_inner` to be able to supply a customized `Budget` w/ a modified signature validation limit. To use the public API we'd need to figure out a way to build a chain that doesn't expend the signature budget before the `build_chain_calls` budget. I haven't been able to think of a way to do that :-/

Reply:
I updated the original test to use the `mini` budget.